server/service/src/item/item.rs

```diff
@@ -49,3 +140,207 @@ pub fn get_item(
+                item_stats,
+                min_months_of_stock,
+                max_months_of_stock,
+            );
+
+            assert_eq!(result, ["item_1".to_string()]);
+        }
+
+        #[test]
+        fn filters_when_min_mos_provided() {
+            let min_months_of_stock = Some(3.0);
+            let max_months_of_stock = None;
+            let mut item_stats = HashMap::new();
+
+            item_stats.insert(
+                "item_1".to_string(),
+                ItemStats {
+                    // This item has 5 MOS on hand, more than min_months_of_stock
+                    average_monthly_consumption: 2.0,
+                    total_stock_on_hand: 10.0,
+                    ..Default::default()
+                },
+            );
+            item_stats.insert(
+                "item_2".to_string(),
+                ItemStats {
+                    // This item has 2 MOS on hand, less than min_months_of_stock
+                    average_monthly_consumption: 5.0,
+                    total_stock_on_hand: 10.0,
+                    ..Default::default()
+                },
+            );
+
+            let result = get_items_ids_for_months_of_stock(
+                item_stats,
+                min_months_of_stock,
+                max_months_of_stock,
+            );
+
+            assert_eq!(result, ["item_1".to_string()]);
+        }
+
+        #[test]
+        fn filters_when_max_mos_provided() {
+            let min_months_of_stock = None;
+            let max_months_of_stock = Some(6.0);
+            let mut item_stats = HashMap::new();
+
+            item_stats.insert(
+                "item_1".to_string(),
+                ItemStats {
+                    // This item has 1 MOS on hand, which is less than the maximum
+                    average_monthly_consumption: 3.0,
+                    total_stock_on_hand: 3.0,
+                    ..Default::default()
+                },
+            );
+
+            item_stats.insert(
+                "item_2".to_string(),
+                ItemStats {
+                    // This item has 6 MOS on hand, which is less than the maximum
+                    average_monthly_consumption: 1.0,
+                    total_stock_on_hand: 6.0,
+                    ..Default::default()
+                },
+            );
+
+            item_stats.insert(
+                "item_3".to_string(),
+                ItemStats {
+                    // This item has 7 MOS on hand, which is more than the maximum
+                    average_monthly_consumption: 1.0,
+                    total_stock_on_hand: 7.0,
+                    ..Default::default()
+                },
+            );
+
+            let result = get_items_ids_for_months_of_stock(
+                item_stats,
+                min_months_of_stock,
+                max_months_of_stock,
+            );
+
+            assert_eq!(result, ["item_2".to_string(), "item_1".to_string()]);
```

Comment:
Ah, actually, looks like this one might be flaky, what order the vec comes back in. Before you merge, could you update this test - either call a sort on the `result`, so you can rely on the order, or could instead do something like assert result length is 2 and item_3 not included 🙏